ecosystem/sep-0010.md

```diff
@@ -139,7 +139,80 @@ operation of the challenge is the **Home Domain**, and that the web auth domain
+
+- `iat` (the time at which the JWT was issued
+  [RFC7519, Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6)) — current timestamp (for example
+  `1530644093`)
+- `exp` (the expiration time on or after which the JWT must not be accepted for processing,
+  [RFC7519, Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4))
+- `web_auth_endpoint` - should match the auth endpoint (`<WEB_AUTH_ENDPOINT>`)
+- All request's parameters (if specified in the request)
+
+Client must then correctly sign the payload with appropriate Stellar private key. To choose the private key client
+application should follow this steps:
+
+- If `client_domain` is specified, the token must be signed with the _Client Domain Account_ (i.e. [SEP-1] defined
+  `SIGNING_KEY`). (See [Verifying the Client Domain](#verifying-the-client-domain) section for more information)
+- Otherwise, the token must be signed with a private key of the **Client Account** passed in the `account` field.
+
+Token then should be signed with the selected private key, using **ed25519** algorithm. An appropriate header must be
+included
+
+```json
+{
+  "alg": "EdDSA"
+}
+```
+
+Example of the valid signed token in base64 format:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJpYXQiOjE3MTEwNzMyMjAsImV4cCI6MTcxMTA3NDEyMCwiYWNjb3VudCI6IkdDNlVDWFZUQU1ORzVKTE9NWkJTQ05ZWFZTTk5GSEwyM1NKUFlPT0ZKRTJBVllERFMyRkZUNDVDIiwiY2xpZW50X2RvbWFpbiI6ImV4YW1wbGUtd2FsbGV0LnN0ZWxsYXIub3JnIiwiaG9zdCI6ImV4YW1wbGUuY29tIiwicGF0aCI6Ii9zZXAxMC9hdXRoIn0.SJwDQBr_xjvb-Nx23t8xhnxzVZVY_IKCzTtPk0-xHfkse7sIyJK_Aobzq3JYpxZ5F5CpNEfOi4Yu5U4mDWVnDA
+```
+
+This token was issued for the request URL
+`https://example.com/sep10/auth?account=GC6UCXVTAMNG5JLOMZBSCNYXVSNNFHL23SJPYOOFJE2AVYDDS2FFT45C&client_domain=example-wallet.stellar.org`
+and signed with the `GCADAIACE6CRWGOB3HJRXIOHMQEUKPNLAUIYFIE26F3OK72RSHGAMARK` key, belonging to a hypothetical domain
+`example-wallet.stellar.org`
+
+Another example of the valid signed token:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJpYXQiOjE3MTEwNzMyMjMsImV4cCI6MTcxMTA3NDEyMywiYWNjb3VudCI6IkdDWFhINkFZSlVWVERHSUhUNDJPWk5NRjNMSENWNERPS0NYNkhIREtXRUNVWllYRFpTV1pONkhTIiwibWVtbyI6IjEyMzQ1NjciLCJob3N0IjoiZXhhbXBsZS5jb20iLCJwYXRoIjoiL3NlcDEwL2F1dGgifQ.wQdzjK2iPiU2C1rpreOEgZalwAdo1nxs2cnD6N1bst4bpwHTnbBE1jU4un0cJjWBMkH_NX3J8gfX082h6VMKDw
+```
+
+The token was issued for the request URL
+`https://example.com/sep10/auth?account=GCXXH6AYJUVTDGIHT42OZNMF3LHCV4DOKCX6HHDKWECUZYXDZSWZN6HS&memo=1234567`. This
+time the token is signed with a **Client Account** () private key
+
+##### Verifying request header
+
+When server receives a request with an `Authorization` header, server should validate the token before processing with
+issuing a challenge transaction.  
+First, server must use the same steps of choosing the public key (see above), and verify **ed25519** signature of the
+token. Next, server must verify JWT claims:
+
+1. `host` must match server's domain
+2. `path` path must match `<WEB_AUTH_ENDPOINT>`
```

Comment:
hummm are we going to use `host` + `path` or just `web_auth_endpoint` ?

Reply:
web_auth_endpoint